### PR TITLE
Feat/aut 1168/e2e asset a block to canvas

### PR DESCRIPTION
**Pull request by Silvia-loza**
Related to: https://oat-sa.atlassian.net/browse/AUT-1168

**Description of changes:**
 E2E tests for Asset. Add A-block to canvas

**How to run locally:**
Checkout to feat/AUT-1168/e2e-Asset-A-block-to-canvas

**run tests from tao core extn:**
npm run cy:open - to run in browser
or npm run cy:run - for headless execution

All tests should pass:

image

### CODE REVIEW

File: views/cypress/tests/assets.spec.js [hunk 1]
```diff
@@ -63,10 +77,32 @@ describe('Assets', () => {
...
+            cy.log('ADD A-BLOCK TO CANVAS');
+            addBlockAndInlineInteractions();
+        });
+        it('can save asset with A-block in it', function () {
+            cy.log('SAVE ASSET WITH A-BLOCK IN IT');
+            cy.intercept('PATCH', '**/taoMediaManager/SharedStimulus/patch*').as('saveAsset');
+            cy.get('[data-testid="save-the-asset"]').click();
+             cy.wait('@saveAsset')
+                 .its('response.body')
+                 .its('success')
```
Comment (adriloma) [inline views/cypress/tests/assets.spec.js:90]: This should work right?
```
     cy.wait('@saveAsset')
                 .its('response.body.success')
```

File: views/cypress/utils/add-A-block-interaction-asset.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Add A-Block for Asset
+ */
+export function addBlockAndInlineInteractions() {
+    cy.log('ADDING A-BLOCK INTERACTION');
+    const blockSelector = `[data-qti-class="${blockContainer}"]`;
+    cy.dragAndDrop(blockSelector, dropSelector);
+    // check that widget is initialized
+    cy.getSettled(`${dropSelector} .widget-box.edit-active${blockSelector}`).should('exist');
+    cy.log('A-BLOCK IS ADDED');
+
```
Comment (adriloma) [inline views/cypress/utils/add-A-block-interaction-asset.js:35]: extra breaklien

**Review by adriloma — APPROVED** [review]
Approved, just two minor comments to consider.
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following conventional commits
- [x] Commits messages are meaningful
- [x] Commits are atomic

File: views/cypress/tests/assets.spec.js [hunk 3]
```diff
@@ -63,10 +77,32 @@ describe('Assets', () => {
...
         it('can go back', function () {
             cy.get(selectors.manageAssets).click();
             cy.getSettled(selectors.treeMediaManager).find(`li[title="${AssetRenamed}"]`);
         });
+        it('can re-open the asset and validates that added block is present', function () {
+            cy.log('VALIDATES THAT A-BLOCK PRESENT WHEN REPOENED');
+            cy.get('[id="media-authoring"]').click();
+            cy.get('.widget-box.widget-block.widget-textBlock').should('exist');
+            cy.get('[data-testid="manage-assets"]').click();
+
```
Comment (adriloma) [inline views/cypress/tests/assets.spec.js:103]: extra breakline here

File: views/cypress/tests/assets.spec.js [hunk 4]
```diff
@@ -110,4 +146,4 @@ describe('Assets', () => {
             );
         });
     });
-});
+});
```
Comment (bziondik) [inline views/cypress/tests/assets.spec.js:149]: empty line

**Review by bziondik — APPROVED** [review]
- [x]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Pull request merged**